wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -122,9 +124,10 @@ public WorkspaceService(
       WorkspaceLinksGenerator linksGenerator,
       @Named(CHE_WORKSPACE_PLUGIN_REGISTRY_URL_PROPERTY) @Nullable String pluginRegistryUrl,
       @Named(CHE_WORKSPACE_DEVFILE_REGISTRY_URL_PROPERTY) @Nullable String devfileRegistryUrl,
-      @Named(CHE_WORKSPACE_PERSIST_VOLUMES_PROPERTY) boolean defaultPersistVolumes,
       URLFetcher urlFetcher,
-      @Named(DEBUG_WORKSPACE_START_LOG_LIMIT_BYTES) Long logLimitBytes) {
+      @Named(DEBUG_WORKSPACE_START_LOG_LIMIT_BYTES) Long logLimitBytes,
+      @Named(CHE_WORKSPACE_STORAGE_AVAILABLE_TYPES) @Nullable String availableStorageTypes,
+      @Named(CHE_WORKSPACE_STORAGE_DEFAULT_TYPE) @Nullable String defaultStorageType) {
```

Comment:
Should it be really nullable? How client should handle missing value?

Reply:
I think yes it can be `null`, client should take care about that. Some client (not dashboard) might not support this feature, it's optional.

Reply:
Let's postpone considering clients which don't support this feature... If it's not supported by client - then client just ignored the value here...

Let's focus on the client which is supposed to use these values - dashboard. 
Do you want to say that dashboard should be adaptive and hide storage type selector if default type and available types are null?

Reply:
Yes, I think so

Reply:
After discussion with @sleshchenko decided to make it not nullable